From 3316e81c2f194ccb247919f1804fa9139ccad596 Mon Sep 17 00:00:00 2001
From: Nikolay Elenkov <nikolayelenkov@google.com>
Date: Wed, 26 Jun 2024 08:01:46 +0000
Subject: [PATCH] Allow system_server to call IKeystoreMaintenance.deleteAllKeys()

This allows RecoverySystem to destroy all synthetic blob protector keys
and make FBE-encrypted data unrecoverable even if data wipe in recovery
is interrupted or skipped.

Bug: 324321147
Test: Manual - System -> Reset options -> Erase all data.
Test: Hold VolDown key to interrupt reboot and stop at bootloader
screen.
Test: fastboot oem bcd wipe command && fastboot oem bcd wipe recovery
Test: fastboot reboot
Test: Device reboots into recovery and prompts to factory reset:
Test: 'Cannot load Android system. Your data may be corrupt. ...
(cherry picked from https://android-review.googlesource.com/q/commit:3941b6874350fb1c8558fcd539ec0ec5038c1d7e)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:72313f580e19af6fbbe95187881c4771a0f2416b)
Merged-In: I5be2f9e8314d36448994f4f14ff585ded7095c8c
Change-Id: I5be2f9e8314d36448994f4f14ff585ded7095c8c
---

diff --git a/prebuilts/api/33.0/private/system_server.te b/prebuilts/api/33.0/private/system_server.te
index 5e826bf..ba71ace 100644
--- a/prebuilts/api/33.0/private/system_server.te
+++ b/prebuilts/api/33.0/private/system_server.te
@@ -970,6 +970,7 @@
 	clear_ns
 	clear_uid
 	get_state
+	delete_all_keys
 	lock
 	pull_metrics
 	reset
diff --git a/private/system_server.te b/private/system_server.te
index 5e826bf..ba71ace 100644
--- a/private/system_server.te
+++ b/private/system_server.te
@@ -970,6 +970,7 @@
 	clear_ns
 	clear_uid
 	get_state
+	delete_all_keys
 	lock
 	pull_metrics
 	reset
